tests/test_inference_quality.py

```diff
@@ -0,0 +1,172 @@
+
+    bdb.execute('''
+        CREATE POPULATION FOR %s WITH SCHEMA (
+            GUESS STATTYPES OF (*)
+        )
+    ''' % (quoted_table,))
+    bdb.execute('CREATE GENERATOR FOR %s USING loom;' % (quoted_table,))
+    bdb.execute('INITIALIZE 4 MODELS FOR %s;' % (quoted_table,))
+    bdb.execute('ANALYZE %s FOR 100 ITERATIONS;' % (quoted_table,))
+
+def insert_row(bdb, table, x, y):
+    quoted_table = bayeslite.bql_quote_name(table)
+    query = '''INSERT INTO {table} ("0", "1") VALUES (?, ?)'''.format(table=quoted_table)
+    bdb.sql_execute(query, bindings=(x, y))
+    return bdb.sql_execute('SELECT last_insert_rowid()').next()[0]
+
+def simulate_from_rowid(bdb, table, column, rowid, limit=1000):
+    quoted_table = bayeslite.bql_quote_name(table)
+    quoted_column = bayeslite.bql_quote_name(str(column))
+    cursor = bdb.execute('SIMULATE {column} FROM {table} GIVEN rowid={rowid} LIMIT {limit}'.format(
+        column=quoted_column,
+        table=quoted_table,
+        rowid=rowid,
+        limit=limit
+    ))
+    return [float(x[0]) for x in cursor]
+
+def test_mix_ratio():
+    means = ((0,20), (20,0))
+    sample_size = 100
+    mix_ratio = [0.7, 0.3]
+    seed = 9999
+    table = 'data'
+
+    sample_gaussians = axis_aligned_gaussians(means, sample_size, seed=seed)
+    samples = mix(sample_gaussians, mix_ratio, seed=seed)
+
+    packed_seed = struct.pack('<QQQQ', seed, seed, seed, seed)
+    with bayeslite.bayesdb_open(seed=packed_seed) as bdb:
+        register_loom(bdb)
+        prepare_bdb(bdb, samples, table, seed)
+
+        cursor = bdb.execute('SIMULATE "0", "1" FROM data LIMIT ?;', (sample_size,))
+        simulated_samples = [sample for sample in cursor]
+
+    counts = collections.Counter(
+        (0 if distance((x,y), means[0]) < distance((x,y), means[1]) else 1
+         for x, y
+         in simulated_samples)
+    )
+    simulated_mix_ratio = [counts[key] / float(len(simulated_samples)) for key in counts]
+
+    for i in range(len(means)):
+        difference = abs(mix_ratio[i] - simulated_mix_ratio[i])
+        assert difference < 0.1
+
+def test_simulate_y_from_partially_populated_row():
+    means = ((0,20), (20,0))
+    sample_size = 50
+    mix_ratio = [0.7, 0.3]
+    seed = 9999
+    table = 'data'
+
+    sample_gaussians = axis_aligned_gaussians(means, sample_size, seed=seed)
+    samples = mix(sample_gaussians, mix_ratio, seed=seed)
+
+    packed_seed = struct.pack('<QQQQ', seed, seed, seed, seed)
+    with bayeslite.bayesdb_open(seed=packed_seed) as bdb:
+        register_loom(bdb)
+        prepare_bdb(bdb, samples, table, seed)
+
+        rowid = insert_row(bdb, table, means[0][0], None)
+        simulated_samples = simulate_from_rowid(bdb, table, 1, rowid, limit=sample_size)
+
+    y_samples = [y for x, y in sample_gaussians[0]]
+    statistic, p_value = stats.ks_2samp(y_samples, simulated_samples)
+    assert(statistic < 0.1 or p_value > 0.01)
```

Comment:
Not sure what values to be asserting on here. I took these values from the [scipy documentation for `ks_2samp`](https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.ks_2samp.html).